### ISSUE TITLE
Terraform & Fortigate Reordering Error 

### ISSUE THREAD

**Issue by leonardoauribe** [fortinetdev/terraform-provider-fortios#224]
Terraform and Fortigate are conflicting on the ordering of members in my service and address groups causing diffs and always appearing in my terraform plans. 

TERRAFORM PLAN
```
  # module.policy_objects.fortios_firewallservice_group.dod_vip_svcgrp[0] will be updated in-place
  ~ resource "fortios_firewallservice_group" "dod_vip_svcgrp" {
        id                    = "dod-vip-ports"
        name                  = "dod-vip-ports"
        # (5 unchanged attributes hidden)

      ~ member {
          ~ name = "MTLS" -> "HTTPS_ALT"
        }
      ~ member {
          ~ name = "HTTPS_ALT" -> "MTLS"
        }
        # (2 unchanged blocks hidden)
    }

  # module.policy_objects.fortios_firewallservice_group.mail_protocols_svcgrp[0] will be updated in-place
  ~ resource "fortios_firewallservice_group" "mail_protocols_svcgrp" {
        id                    = "mail-protocols"
        name                  = "mail-protocols"
        # (5 unchanged attributes hidden)

      ~ member {
          ~ name = "SMTP_10025" -> "SMTP_587"
        }
      ~ member {
          ~ name = "SMTP_11025" -> "SMTP_10025"
        }
      ~ member {
          ~ name = "SMTP_587" -> "SMTP_11025"
        }
        # (1 unchanged block hidden)
    }

'''

The Fortigate takes the configuration and reorders it once applied. I keep getting the same output shown above over and over again. This is an issue because this code is applied to multiple Fortigates some of which have the ordering slightly different so If I fix it in code for one it will likely cause the issue in another. 

Is there a way for terraform to ignore the ordering or some other solution?

**Comment by lix-fortinet** [comment]
Hi @leonardoauribe,

Thank you for raising this issue. We will appreciate it if you could provide an example of your test case and the environment for your test case, like the FortiOS version, FortiOS Terraform provider version, Terraform version, etc. That would be very helpful for us to reproduce this issue.

Thanks,
Xing

**Comment by leonardoauribe** [comment]
Version Details 

```
Terraform v1.0.1
on darwin_amd64
+ provider registry.terraform.io/fortinetdev/fortios v1.14.0
+ provider registry.terraform.io/hashicorp/local v2.2.2

FortiOS version: v6.4.7
```

**Comment by leonardoauribe** [comment]
Argument 'dynamic_sort_subtable" is set to false on all resources. I went ahead and changed the member sequence on the Fortigates to match the sequence in the .tfstate file and that did the trick. I do still believe a solution should be put in place to where order doesn't matter. thanks for the help on this issue.

**Comment by lix-fortinet** [comment]
Hi @leonardoauribe,

Thank you for your comment. Team are working on the sorting issue. We will get back to you once we have any updates.

Thanks,
Xing

**Comment by leonardoauribe** [comment]
It appears that this issue does still exist on some of our Fortigates. Has there been any progress made with fixing the sorting issue?

**Comment by kunstkomputer** [comment]
I can confirm, that we are seeing the same issue with another ressource. We are trying to manage `fortios_firewall_addrgrp` with roughly 50 members. We also observe, that objects are reordered by the provider or by API itself. 
Is there anything we can do in this case, we would be also happy to contribute.

**Comment by lix-fortinet** [comment]
Hi,

Sorry that we forgot to update this issue. We made improvement when `dynamic_sort_subtable` set to true. If set  `dynamic_sort_subtable` to true, user need to make sure the order on .tf file are sorted. For example, could use Terraform function `sort` sort the items:
```
resource "fortios_firewall_addrgrp" "dummy_address_group" {
  name   = "dummy_address_group"
  dynamic_sort_subtable = true

  dynamic "member" {
    for_each = sort([ fortios_firewall_address.trname[2].name,
                 fortios_firewall_address.trname[0].name, 
               fortios_firewall_address.trname[1].name ])
    content { 
      name = member.value
    }
  }
}
```

As for setting `dynamic_sort_subtable` to false, the provider will send the request with the order in .tf file. But the REST API did not reorder the list variable for PUT operation. We will talk with REST API team about this issue. 

Thanks,
Xing

**Comment by leonardoauribe** [comment]
Thanks for the update! Right now we have "dynamic_sort_subtable" set to false. I'll wait to see what you hear back from REST API team.

**Comment by vincent-at-forti** [comment]
Hello Xing,

Thanks for your insight.
Just a question: Members in firewall addrgrp, firewall service custom group , user groups etc... should be unordered as we don't really care of orders for them.
I believe the schema definition of these objects should be more :
schema.TypeSet (https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-types#typeset ) which are unordered
instead of 
schema.TypeList ( https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-types#typelist ) which are ordered.

And in that case , we could get rid of that reordering order as terraform would not care of order while planning the change.

Or am I misleading?

**Comment by lix-fortinet** [comment]
Hi Vincent,

Thank you for your suggestion! We will investigate it and improve it on next release.

Thanks,
Xing

**Comment by leonardoauribe** [comment]
Any update on this?

**Comment by MaxxLiu22** [comment]
Hi @leonardoauribe ,

Sorry for the late update, this blocks reorder issue has been fixed on the latest terraform FOS provider 1.18.1, feel free to upgrade to it.  Let me know if that doesn't solve your question.

Thanks,
Maxx

**Comment by leonardoauribe** [comment]
thanks for the update! I will to get this provider upgraded.
